Document multiple level rolling upgrade [MC-2820]. (#328)
hazelcast/management-center-docs: docs/modules/clusters/pages/triggering-rolling-upgrade.adoc

+
image:ROOT:RollingUpgrade.png[RollingUpgradeMenu]

+
[NOTE]
====
Management Center can offer the following types of rolling upgrades:

* From version 4.2 to 5.0.

> * From any version earlier than 5.x, excluding 4.2, to the next minor version. For example, from version 4.1 to 4.2.

Reviewer: `From any version earlier than 5.x` also `any version from 4.0`. Upgrade from 3.x to 4.0 is not supported because of breaking changes

> * For versions below 5.x and other than 4.2: Upgrade will be made to the next minor version. Example: For version 4.1, upgrade will be to 4.2.

Reviewer 2: ```suggestion
* From any version earlier than 5.x, excluding 4.2, to the next minor version. For example, from version 4.1 to 4.2
```

> * If members are already version 5.5, you can upgrade your cluster across multiple versions directly to the 5.5 long-term support version from any 5.x version. If any member version is earlier than 5.5, the cluster can only be upgraded to the next minor version.

Reviewer: do we have a definition of `long-term support version` in teh Platform docs? I think it would be useful to add a link here

Reviewer 2: @promanenko we will have - the page is not yet on the 5.5, but the [PR ](https://github.com/hazelcast/hz-docs/pull/1103) is almost there; I'll get the latest comments addressed today and try to progress it; you should be able to add the link in advance (just ignore the build errors)

> * For versions above and equal to 5.x: Since version 5.5 is Long Term Support, multiple version upgrade is possible. Example: If the cluster version is 5.3 and all member versions are equal and above 5.5, the upgrade will be made to 5.5. If any member version is below than 5.5, upgrade will be made to the next minor version.

Reviewer 2: ```suggestion
* From any version of 5.x or later, you can upgrade across multiple versions directly to the long-term support 5.5 version. For example, if the cluster version is 5.3 and all member versions are at least 5.5, you can upgrade directly to 5.5. If any member version is earlier than 5.5, upgrade is to to the next minor version.
```

Reviewer 2: I am unsure that this is totally correct - unless the behaviour is exclusive to MC - for the following reasons:

- If you have a 5.3 cluster with all members at 5.5, you can directly upgrade the cluster to 5.5
- If you have a member version that is earlier than 5.5 (but still 5.x), you can upgrade the member, then use LTS upgrade to upgrade the cluster
- If you have a version later than 5.5, you are into the world of STS support - at this stage, it is expected that you will be able to either upgrade to each STS until the next LTS, or directly from one LTS to the next. I'd recommend for this release we do not mention the behaviour of anything later than 5.5, as it is not yet set in stone.

@frant-hartm can you confirm I have correctly represented this behaviour, please

Author: i think:
item 1 is aligned with what is written.
item 2 -> if we have a member version that is earlier than 5.5 (but still 5.x) --> this case already represents a scenario. right at this point MC offers 5.x+1.. 
* clients can chose to not upgrade it and apply what you wrote: `you can upgrade the member, then use LTS upgrade to upgrade the cluster`. if they upgrade the member, then this scenario falls into category of item 1.

item 3 -> ` I'd recommend for this release we do not mention the behaviour of anything later than 5.5` i agree. `For versions above and equal to 5.x` may create confusion. 
what about: `From any version between (inclusive) 5.0 or 5.x`, you can upgrade across multiple versions directly to the long-term support 5.5 version. For example, if the cluster version is 5.3 and all member versions are at least 5.5, you can upgrade directly to 5.5. If any member version is earlier than 5.5, upgrade is to to the next minor version.

wdyt?

Reviewer 2: How about, turning it on its head slightly:

If members are already version 5.5, you can upgrade your cluster across multiple versions directly to the 5.5 long-term support version from any 5.x version. If any member version is earlier than 5.5, the cluster can only be upgraded to the next minor version.

?

It is a gnarly thing to explain in few words - with member versions and cluster versions to consider as well as LTS versions and the compatibilites. I have attempted to draw it out slightly, but let me know if it hasn't worked!

Author: ```
If members are already version 5.5, you can upgrade your cluster across multiple versions directly to the 5.5 long-term support version from any 5.x version. If any member version is earlier than 5.5, the cluster can only be upgraded to the next minor version.
```
looks good, thanks for the suggestions!
